application/common/components/adapters/SagePeopleIdStore.php

```diff
@@ -0,0 +1,284 @@
+class SagePeopleIdStore extends IdStoreBase
+{
+    const PROP_EMPLOYEE_ID = 'fHCM2__Unique_Id__c';
+    const PROP_FIRST_NAME = 'fHCM2__First_Name__c';
+    const PROP_LAST_NAME = 'fHCM2__Surname__c';
+    const PROP_DISPLAY_NAME = 'Name';
+    const PROP_PERSONAL_EMAIL = 'fHCM2__Home_Email__c';
+    const PROP_EMAIL = 'fHCM2__User__r.Email';
+    const PROP_USERNAME = 'fHCM2__User__r.Username';
+    const PROP_MANAGER_EMAIL = 'fHCM2__Manager_User__r.Email';
+    const PROP_LOCKED = 'fHCM2__User__r.jaars_Locked_From_IdP__c';
+    const PROP_REQUIRE_MFA = 'fHCM2__User__r.jaars_Require_2sv_with_IdP__c';
+
+    public $authUrl = null;
+    public $queryUrl = null;
+    public $clientId = null;
+    public $clientSecret = null;
+    public $username = null;
+    public $password = null;
+
+    public $timeout = 45; // Timeout in seconds (per call to ID Store API).
+
+    protected $httpClient = null;
+
+    public function init()
+    {
+        $requiredProperties = [
+            'authUrl',
+            'queryUrl',
+            'clientId',
+            'clientSecret',
+            'username',
+            'password',
+        ];
+        foreach ($requiredProperties as $requiredProperty) {
+            if (empty($this->$requiredProperty)) {
+                throw new InvalidArgumentException(sprintf(
+                    'No %s was provided.',
+                    $requiredProperty
+                ), 1559070540);
+            }
+        }
+
+        parent::init();
+    }
+
+    /**
+     * Return an associative array with key being the Sage People field names
+     * and values being the ID Broker field names.
+     * @return array
+     */
+    public static function getIdBrokerFieldNames()
+    {
+        return [
+            self::PROP_EMPLOYEE_ID => User::EMPLOYEE_ID,
+            self::PROP_FIRST_NAME => User::FIRST_NAME,
+            self::PROP_LAST_NAME => User::LAST_NAME,
+            self::PROP_DISPLAY_NAME => User::DISPLAY_NAME,
+            self::PROP_PERSONAL_EMAIL => User::PERSONAL_EMAIL,
+            self::PROP_EMAIL => User::EMAIL,
+            self::PROP_USERNAME => User::USERNAME,
+            self::PROP_MANAGER_EMAIL => User::MANAGER_EMAIL,
+            self::PROP_LOCKED => User::LOCKED,
+            self::PROP_REQUIRE_MFA => User::REQUIRE_MFA,
+            // No 'active' needed, since all ID Store records returned are active.
+        ];
+    }
+
+    /**
+     * Get the specified user's information. Note that inactive users will be
+     * treated as non-existent users.
+     *
+     * @param string $employeeId The Employee ID.
+     * @return User|null Information about the specified user, or null if no
+     *     such active user was found.
+     * @throws Exception
+     */
+    public function getActiveUser(string $employeeId)
+    {
+        $activeUsers = $this->getFromIdStore(sprintf(
+            "WHERE %s='%s' AND fHCM2__Employment_Status__c='Active'",
+            self::PROP_EMPLOYEE_ID,
+            $employeeId
```

Comment:
Are you talking about doing something like calling addslashes() on the constant and the variable? 

I understand the risk of programmatic creation of SQL statements, but isn't that only when user-supplied data is involved?

Reply:
Just realized that `$employeeId` could be considered user-supplied since it comes from the API. Depending on whether the single-user update capability is even needed, I could just remove this code.

Reply:
You know, since the Sage People adapter doesn't support `syncUsersChangedSince()`, I think it probably doesn't need to support `getActiveUser()`, so you probably _could_ simply remove this method.

I'm fairly sure `getActiveUser()` is only used when (1) syncing a specific user or (2) syncing the specific users changed since a particular point in time. If Sage People isn't going to make any call to the ID Sync telling it to sync a specific user who was just changed, and if it doesn't provide a way for ID Sync to get the list of users changed since a specific time, then `getActiveUser()` will probably never be called.

Reply:
I think I'll leave it since I don't know ~for sure~ that it wouldn't be used, but will ensure it's protected from SQL injection. (To be continued on PR #88)